lib/rubocop/cop/lint/duplicate_regexp_character_class_element.rb

```diff
@@ -75,9 +93,12 @@ def within_interpolation?(node, child)
           interpolation_locs(node).any? { |il| il.overlaps?(parse_tree_child_loc) }
         end
 
-        def start_with_escaped_zero_number?(current_child, next_child)
-          # Represents escaped code from `"\00"` (`"\u0000"`) to `"\07"` (`"\a"`).
-          current_child == '\\0' && next_child.match?(/[0-7]/)
+        def escaped_octal?(string)
+          string.size == 2 && string[0] == '\\' && octal?(string[1])
```

Comment:
That's just my two cents:

```suggestion
          string.length == 2 && string[0] == '\\' && octal?(string[1])
```